pkg/cluster/sync.go

```diff
@@ -569,6 +588,16 @@ func (c *Cluster) syncRoles() (err error) {
 		return fmt.Errorf("error getting users from the database: %v", err)
 	}
 
+	// update pgUsers where a deleted role was found
+	// so that they are skipped in ProduceSyncRequests
+	for _, dbUser := range dbUsers {
+		if originalUser, exists := deletedUsers[dbUser.Name]; exists {
+			recreatedUser := c.pgUsers[originalUser]
+			recreatedUser.Deleted = true
+			c.pgUsers[originalUser] = recreatedUser
```

Comment:
what about sth like
```golang
c.pgUsers[originalUser].Deleted = true
```
? Just to elimiante the extra variable "recreatedUser"

Reply:
`cannot assign to struct field c.pgUsers[originalUser].Deleted in map`